pkg/api/types.go

```diff
@@ -921,6 +921,9 @@ type PodSpec struct {
 	// used must be specified.
 	// Optional: Default to false.
 	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// Run the container using the host ipc mode.
+	// Optional: Default to false.
+	HostIPC bool `json:"hostIPC,omitempty"`
```

Comment:
I think we do.  There are lots of cases where we want to run daemon
controllers that *only* have access to the host network - sharing host IPC
or host mount namespace or host PID is wrong.

On Thu, Aug 13, 2015 at 1:23 PM, Vish Kannan <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/12470#discussion_r37000555>
> :
>
> > @@ -921,6 +921,9 @@ type PodSpec struct {
> >  	// used must be specified.
> >  	// Optional: Default to false.
> >  	HostNetwork bool `json:"hostNetwork,omitempty"`
> > +	// Run the container using the host ipc mode.
> > +	// Optional: Default to false.
> > +	HostIPC bool `json:"hostIPC,omitempty"`
>
> Do we need a per namespace knob in the API? Can we instead have a single
> privileged option?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12470/files#r37000555>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift